zpay32/invoice.go

```diff
@@ -287,21 +290,32 @@ func Decode(invoice string) (*Invoice, error) {
 	// address. This will also determine which network this invoice is
 	// meant for.
 	var net *chaincfg.Params
-	if strings.HasPrefix(hrp[2:], chaincfg.MainNetParams.Bech32HRPSegwit) {
+	switch {
+	case strings.HasPrefix(hrp[2:], chaincfg.MainNetParams.Bech32HRPSegwit):
 		net = &chaincfg.MainNetParams
-	} else if strings.HasPrefix(hrp[2:], chaincfg.TestNet3Params.Bech32HRPSegwit) {
+	case strings.HasPrefix(hrp[2:], chaincfg.TestNet3Params.Bech32HRPSegwit):
 		net = &chaincfg.TestNet3Params
-	} else if strings.HasPrefix(hrp[2:], chaincfg.SimNetParams.Bech32HRPSegwit) {
+	case strings.HasPrefix(hrp[2:], chaincfg.SimNetParams.Bech32HRPSegwit):
 		net = &chaincfg.SimNetParams
-	} else {
+
+	// For now, litecoin network prefixes use bitcoin config with
+	// modified configuration parameters.
```

Comment:
Add TODO that this should be changed.

Reply:
@halseth Should I mention that the desired approach uses an interface for chain configuration params? I'd be happy to assign that to myself and start working on it if that's alright.

Reply:
Yep, awesome!